ckanext/datatablesview/config_declaration.yaml

```diff
@@ -93,3 +93,9 @@ groups:
       the user to view, copy and print the details of a specific row. This
       value can be overridden at the resource level when configuring a
       DataTables resource view.
+
+  - key: ckan.datatables.null_label
+    example: N/A
+    description: |
+      The option defines the label used to display NoneType values for the front-end.
+      This should be a string and can be translated via po files.
```

Comment:
can we add the new default value here too?

Reply:
@wardi default value now added. I set it as `''` but should it just be blank?

Are we able to add `type: string` or `type: str` here too? I have not seen any other declaration file use type string?

Reply:
Yes, exactly. `type` defines some basic conversion rules. There is no such `type` as `string`, but if it were there, it would just add a `unicode_safe` or `unicode_only` validator. Such type(and these validators used by themselves, without any additional validators) have no practical sense inside config declaration. Validators that are applied in the following cases:

* When value read from config file. All the values initially are strings because of the internal logic of `configparser` library, which handles INI files. It means that `unicode_only` won't do much here
* When the value is missing from the config file, but declared `default` is available. If you expect the value to be a string, you should set the default to string. So here we don't need the validator as well.
* When the value is missing from the config file and the declaration has no `default`. In this case, we are using `None` as a default `default`(sounds awesome). I don't think that applying `unicode_*` validators to `None` is a great idea. The only case, when it may be useful - required options. You set no default and add a `unicode_only` validator. Then you'll see a validation error because `None` won't pass this check. But the much better way to handle it is using a `not_empty` validator, so `unicode_*` validators are not needed here as well


I thought about adding `str` for completeness. But if a type that doesn't do much exists, it has sense to add more useful types, like `dict`, `enum`, `float`, `set`, `timedelta`, `Path`, etc. So, for the first version, I decided to go with the minimal required set of types. And, when we get some feedback from the real world, we can define hundreds of types for every possible use case. But remember, that `type` is just a shortcut for `validators: ...` and don't overestimate its value

ℹ️ [here is the list of all available types](https://github.com/ckan/ckan/blob/master/ckan/config/declaration/load.py#L40-L46)

Reply:
Okay, so we do not need a type here in our declaration. Sounds good!